src/storage/txn/commands/atomic_store.rs

```diff
@@ -41,10 +49,27 @@ impl CommandExt for RawAtomicStore {
 }
 
 impl<S: Snapshot, L: LockManager> WriteCommand<S, L> for RawAtomicStore {
-    fn process_write(self, _: S, _: WriteContext<'_, L>) -> Result<WriteResult> {
+    fn process_write(self, snapshot: S, wctx: WriteContext<'_, L>) -> Result<WriteResult> {
+        if self.api_version == ApiVersion::V2 && !snapshot.ext().is_max_ts_synced() {
+            return Err(ErrorInner::MaxTimestampNotSynced {
+                region_id: self.ctx.get_region_id(),
+                start_ts: TimeStamp::zero(),
+            }
+            .into());
+        }
+
+        let provider = self.causal_ts_provider.clone();
+        let concurrency_manager = wctx.concurrency_manager;
+        let lock_guard = block_on(get_raw_key_guard(&provider, concurrency_manager)).map_err(
```

Comment:
Why block?

Reply:
Because `get_raw_key_guard` is a async func.

Reply:
I don't think it's a good implementation. Blocking the thread can starve all the other tasks.

Reply:
Thanks for such a good suggestion, I removed block by removing `asynchronous operation` from atomic_store's `process_write` to scheduler's `[process_write](https://github.com/zeminzhou/tikv/blob/new-flush-for-causalts/src/storage/txn/scheduler.rs#L859)` . PTAL~